code/iaas/simple-allocator/src/main/java/io/cattle/platform/simple/allocator/dao/impl/SimpleAllocatorDaoImpl.java

```diff
@@ -59,7 +55,7 @@
         Map<Long, String> hostIdsToUuids = new HashMap<>();
         if (orderedHostUuids == null) {
             Result<Record3<String, Long, Long>> result = getHostQuery(orderedHostUuids, options).fetch();
-
+            Collections.shuffle(result);
```

Comment:
Randomizing the results so that we don't always pick the same host.  Waddaya think?